src/lib/web-component/filter-option/filter-option-element.js

```diff
@@ -85,6 +84,29 @@ class FilterOption extends HTMLElement {
     this.toggleBtn.classList.toggle('expanded');
   }
 
+  onCheckboxChange(checkboxLabel, isChecked) {
+    if (isChecked) {
+      this.activeCheckboxLabels.add(checkboxLabel);
+    } else {
+      this.activeCheckboxLabels.delete(checkboxLabel);
+    }
+    this.dispatchFilterChangeEvent({ isNewFilterAdded: false });
+  }
+
+  dispatchFilterChangeEvent({ isNewFilterAdded }) {
+    const filterDetail = {
+      filterObject: {
+        logKey: this.filterKey,
+        filterDescriptor: new Set([...this.activeCheckboxLabels]),
```

Comment:
The `Set` constructor takes an iterable. A `Set` instance itself is also iterable, so you don't need to turn it into an array first.

`[...this.activeCheckboxLabels]` iterates over the set and results in an array.
`new Set([...this.activeCheckboxLabels])` results in two iterations: first to turn the set into an array, and then again the `Set` constructor iterates over the array to fill the set.

Reply:
done :-)